packages/devtools_server/lib/src/memory_profile.dart

```diff
@@ -331,6 +330,21 @@ class MemoryProfile {
     }
     return heapList;
   }
+
+  static String _formatTime(DateTime value) {
+    String toStringLength(int value, int length) {
+      final result = '$value';
+      return '0' * (length - result.length) + result;
```

Comment:
I'm not familiar with this syntax. When length - result.length is negative, what will this look like?

Take the example value = 123456, length = 3. What does this method yield?

Reply:
We call this function for values that have less digits than `length`.
I will add `assert`.

Reply:
instead of asserting, I think we need to handle all cases:
```dart
if (length == result.length) {
  return result;
}
if (length > result.length) {
 return '0' * (length - result.length) + result;
}
return result.substring(0, length);
```

Reply:
OK, although we know that we never have more than 24 hours, 60 minutes, and 60 seconds. So, never longer than 2 digits. But this is your code, so this is fine with me.

Reply:
Ah I see what you are saying. In that case you can revert to what you had. Apologies for the churn.